tsan_env_suppressions

```diff
@@ -0,0 +1,8 @@
+# For ThreadSanitizerSuppressions
+# https://github.com/google/sanitizers/wiki/ThreadSanitizerSuppressions
+
+race:*/ring_buffer.h
+
+deadlock:*/store.h
+deadlock:*/untyped_map.h
+deadlock:*/apply_changes.h
```

Comment:
Currently I am working on the partition test, so the suppressions are written to suppress only errors on that test.
First I'll try to fix the errors and remove the suppressions (hopefully this process resolve the failure of MultiThreading CI).
After partition test becomes error free, we should work on the same process for other tests.